### PR TITLE
feat: SCRUM-157-FE-Route

### PR DESCRIPTION
**Pull request by asdelay**
## Describe your changes

- Modified custom calendar now selected date is accessible outside the component
- Fixed bug (when page reloads bottom navbar sets second page as active)
- Created UI for routes page

## Issue ticket code (and/or) and link

- [SCRUM-157](https://codecraftersintership.atlassian.net/browse/SCRUM-157)

### **General**
Screenshot 2024-12-11 at 18 45 06

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Review by AlexeyBezrebriy — APPROVED** [review]

File: src/components/CustomCalendar/useCalendar.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { addMonths, format, subMonths } from 'date-fns';
-import { ReactElement, useState } from 'react';
+import { Dispatch, ReactElement, SetStateAction, useState } from 'react';
```
Comment (Parle648) [inline src/components/CustomCalendar/useCalendar.tsx:1]: import from react to the 1 line

File: src/components/CustomCalendar/useCalendar.tsx [hunk 2]
```diff
@@ -8,7 +8,13 @@ import { StyledWeekday } from './styles';
...
 
-const useCalendar = (): {
+const useCalendar = ({
+  currentDate,
+  setCurrentDate,
+}: {
+  currentDate: Date;
+  setCurrentDate: Dispatch<SetStateAction<Date>>;
+}): {
   currentDate: Date;
```
Comment (Parle648) [inline src/components/CustomCalendar/useCalendar.tsx:13]: I think it's ok when we have 2 or 3 props, I'd better create interface for returned type

File: src/components/CustomCalendar/useCalendar.tsx [hunk 3]
```diff
@@ -19,13 +25,6 @@ const useCalendar = (): {
...
   handleDateChange: (newDate: Value) => void;
 } => {
-  const [currentDate, setCurrentDate] = useState(() => {
-    const today = new Date();
-
-    today.setHours(0, 0, 0, 0);
-
-    return today;
-  });
   const [isExpanded, setIsExpanded] = useState(false);
```
Comment (Parle648) [inline src/components/CustomCalendar/useCalendar.tsx]: add types for useState

File: src/pages/OrderPage/index.tsx [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { Typography } from '@mui/material';
 import { t } from 'i18next';
-import { ReactNode } from 'react';
+import { ReactNode, useState } from 'react';
```
Comment (Parle648) [inline src/pages/OrderPage/index.tsx]: react to the 1 line

File: src/pages/RoutesPage/components/RouteItem/index.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
+import { Typography } from '@mui/material';
+import { format } from 'date-fns';
+import { t } from 'i18next';
+import { FC } from 'react';
```
Comment (Parle648) [inline src/pages/RoutesPage/components/RouteItem/index.tsx]: same

File: src/pages/RoutesPage/components/RouteItem/index.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+import {
```
Comment (Parle648) [inline src/pages/RoutesPage/components/RouteItem/index.tsx:7]: 1.1.14

File: src/pages/RoutesPage/components/RouteItem/index.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+}) => {
+  const idPadding = 6;
+  const timePadding = 2;
+
+  return (
+    <StyledRouteItem>
+      <StyledRouteInfo>
+        <StyledRouteInfoItem>
+          <Typography variant='body2'>{t('routes.route')}</Typography>
+          <Typography sx={{ color: COLORS.schemes.outline }}>
```
Comment (Parle648) [inline src/pages/RoutesPage/components/RouteItem/index.tsx]: 1.3.4

File: src/pages/RoutesPage/components/StatusComponent/index.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import { Typography } from '@mui/material';
+import { t } from 'i18next';
+import { FC } from 'react';
```
Comment (Parle648) [inline src/pages/RoutesPage/components/StatusComponent/index.tsx]: first line

File: src/pages/RoutesPage/components/StatusComponent/index.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+import { StatusItemWrapper } from './styles';
+import { assignColor, assignRouteStatus } from './useStatus';
+
+import { StatusEnum } from '@/constants/status';
```
Comment (Parle648) [inline src/pages/RoutesPage/components/StatusComponent/index.tsx:9]: 1.1.14

File: src/pages/RoutesPage/components/StatusComponent/index.tsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+interface StatusItemProps {
+  status: StatusEnum;
+}
+
+const StatusComponent: FC<StatusItemProps> = ({ status }) => {
+  return (
+    <StatusItemWrapper>
+      <Typography variant='body2' sx={{ color: assignColor(status) }}>
```
Comment (Parle648) [inline src/pages/RoutesPage/components/StatusComponent/index.tsx:18]: 1.3.4

File: src/pages/RoutesPage/index.tsx [hunk 11]
```diff
@@ -0,0 +1,80 @@
+import { t } from 'i18next';
+import { FC, useState } from 'react';
+
+import RouteItem from './components/RouteItem';
+import { OrdersContainer } from './styles';
```
Comment (Parle648) [inline src/pages/RoutesPage/index.tsx:6]: 1.1.14

File: src/components/CustomCalendar/useCalendar.tsx (hunk 1 above)
Comment (asdelay) [reply src/components/CustomCalendar/useCalendar.tsx:1]: Fixed

File: src/components/CustomCalendar/useCalendar.tsx (hunk 2 above)
Comment (asdelay) [reply src/components/CustomCalendar/useCalendar.tsx:13]: Fixed

File: src/components/CustomCalendar/useCalendar.tsx (hunk 3 above)
Comment (asdelay) [reply src/components/CustomCalendar/useCalendar.tsx]: Fixed

File: src/pages/OrderPage/index.tsx (hunk 4 above)
Comment (asdelay) [reply src/pages/OrderPage/index.tsx]: Fixed

File: src/pages/RoutesPage/components/RouteItem/index.tsx (hunk 5 above)
Comment (asdelay) [reply src/pages/RoutesPage/components/RouteItem/index.tsx]: Fixed

File: src/pages/RoutesPage/components/RouteItem/index.tsx (hunk 7 above)
Comment (asdelay) [reply src/pages/RoutesPage/components/RouteItem/index.tsx]: Fixed

File: src/pages/RoutesPage/components/StatusComponent/index.tsx (hunk 8 above)
Comment (asdelay) [reply src/pages/RoutesPage/components/StatusComponent/index.tsx]: Fixed

File: src/pages/RoutesPage/components/StatusComponent/index.tsx (hunk 10 above)
Comment (asdelay) [reply src/pages/RoutesPage/components/StatusComponent/index.tsx:18]: fixed

**Review by Parle648 — APPROVED** [review]

**Review by oyakovytskyi — APPROVED** [review]

**Pull request merged**